### PR TITLE
add a separate "with-jest-jsdom" example

### LINKED ISSUE #99: Separate with-jest and with-jest-and-jsdom examples
Running MSW 2.0 on Jest when JSDOM is not used allows us to skip patching undici. For that reason, we should show separate examples of setting up MSW on Jest with and without JSDOM

### PR DESCRIPTION
**Pull request by VanTanev**
Jest without JSDOM is simpler to run and does not require undici patching.

Closes #99

### CODE REVIEW

File: examples/with-jest/jest.config.ts [hunk 1]
```diff
@@ -3,24 +3,7 @@ import type { Config } from 'jest'
 export default {
   rootDir: __dirname,
   setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
```
Comment (kettanaito) [inline examples/with-jest/jest.config.ts:5]: This example doesn't really need a setup anymore.

File: examples/with-jest/package.json [hunk 2]
```diff
@@ -9,7 +9,6 @@
     "@types/jest": "^29.5.1",
     "@types/node": "^18",
     "jest": "^29.5.0",
-    "jest-environment-jsdom": "^29.5.0",
     "msw": "2.0.14",
     "ts-node": "^10.9.1",
     "typescript": "^5.0.4",
```
Comment (kettanaito) [inline examples/with-jest/package.json:14]: Can remove `undici` from `with-jest` example too. No polyfills needed.

File: examples/with-jest-and-jsdom/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "../../tsconfig.json",
```
Comment (kettanaito) [inline examples/with-jest-and-jsdom/tsconfig.json:2]: What is it intended to extend? Looks like it references a file that lives outside of the example. Let's keep all the examples self-contained.

File: examples/with-jest/example.test.ts [hunk 4]
```diff
@@ -1,7 +1,3 @@
-/**
- * @jest-environment node
- */
-
 it('receives a mocked response to a REST API request', async () => {
```
Comment (kettanaito) [inline examples/with-jest/example.test.ts:1]: The `with-jest` example is missing the MSW integration.

Comment (kettanaito) [inline examples/with-jest/example.test.ts:1]: The `with-jest` example is missing a README.

File: examples/with-jest-and-jsdom/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Jest + JSDOM + MSW
+
+[Mock Service Worker](https://github.com/mswjs/msw) usage example with [Jest](https://github.com/jestjs/jest) and [JSDOM](https://github.com/jsdom/jsdom).
+
+Please examine [jest.config.ts](./jest.config.ts) and [jest.polyfills.ts](./jest.polyfills.ts) carefully!
```
Comment (kettanaito) [inline examples/with-jest-and-jsdom/README.md:5]: I will help rewrite this a bit.

**Comment by kettanaito** [comment]
Thanks for opening this, @VanTanev! I left a few comments, will address them once I have a minute.

File: examples/with-jest-and-jsdom/tsconfig.json (hunk 3 above)
Comment (VanTanev) [reply examples/with-jest-and-jsdom/tsconfig.json:2]: This is the same code as what is in examples/with-jest.

File: examples/with-jest/jest.config.ts (hunk 1 above)
Comment (VanTanev) [reply examples/with-jest/jest.config.ts:5]: It does need it. The setup file sets up MSW globaly, as well as resetHandlers() after each test.

File: examples/with-jest/example.test.ts (hunk 4 above)
Comment (VanTanev) [reply examples/with-jest/example.test.ts:1]: Sorry, I don't understand? It's in the setup file.

Comment (VanTanev) [reply examples/with-jest/example.test.ts:1]: I've kept the original README

File: examples/with-jest/package.json (hunk 2 above)
Comment (VanTanev) [reply examples/with-jest/package.json]: Done

File: examples/with-jest/example.test.ts (hunk 4 above)
Comment (kettanaito) [reply examples/with-jest/example.test.ts:1]: I think GitHub screwed up the diff view for me. Let me checkout this branch and look at the files proper.

Comment (kettanaito) [reply examples/with-jest/example.test.ts:1]: Once again, GitHub shenanigans.

File: examples/with-jest/jest.config.ts (hunk 1 above)
Comment (kettanaito) [reply examples/with-jest/jest.config.ts:5]: You're right!

**Comment by kettanaito** [comment]
I renamed the example, added it to the CI, elaborated more on what and why to do in its README. This should be good now, let's see the tests!

**Comment by VanTanev** [comment]
Oops, I guess removing ts-node was too optimistic - although it did run OK locally?

**Comment by kettanaito** [comment]
@VanTanev, it ran okay locally because other examples used `ts-node`, I suppose. On CI, each example only installs its own dependencies.

**Review by kettanaito — APPROVED** [review]
Fantastic work! Thank you, @VanTanev

**Comment by VanTanev** [comment]
@kettanaito Thank you!

**Pull request merged**